### ISSUE TITLE
Improve LivenessProbe and ReadinessProbe 

### ISSUE THREAD

**Issue by allamand** [Orange-OpenSource/casskop#129]
# Situation
When the Cassandra is under heavy load, we can see lots of java process in Cassandra Pods which are there only to run `nodetool status` for both liveness and readiness health checks.

We also see the case where the pods would be killed by liveness checks while it was still booting.

So we wants to change the way we uses liveness and readiness checks, and also allow users to change thoses scripts with a ConfigMap.

Here are the different proposal we have for thoses checks:

## Liveness

We propose to create new liveness script which will have for default:
 a nodetool info that checks for **Native Transport active: true**

ex:
```
nodetool info | grep "Native Transport active: true"
```

## Readiness probe

We propose to create new readiness script which will have for default:

Since Casskop will first create cluster with default user Cassandra, we would wants CassKop to create monitoring user such as cassandra_monitoring while in run.

cql file command.cql:
```
SHOW VERSION;
exit;
```
script:
```
CQLSH_OUTPUT=$(/usr/bin/cqlsh -u'cassandra' -p'cassandra' -f /etc/liveness/command.cql 2>&1)
if [[ $? -eq 0 ]]; then
    if [[ $ CQLSH _OUTPUT == *" Cassandra 3. "* ]]; then
        ALIVE = true
    else
        CQLSH_OUTPUT/usr/bin/cqlsh -u'pns_cassandra_monitoring' -p'XXXXX' -f /etc/liveness/command.cql 2>&1
        if [[ $? -eq 0 ]]; then
            if [[ $ CQLSH _OUTPUT == *" Cassandra 3. "* ]]; then
                ALIVE = true
            else
                ALIVE =false
                fi
            else
                ALIVE =false
        fi
    fi
else
    CQLSH_OUTPUT=$(/usr/bin/cqlsh -u'pns_cassandra_monitoring' -p'XXXXX' -f /etc/liveness/command.cql 2>&1)
    if [[ $? -eq 0 ]]; then
        if [[ $ CQLSH _OUTPUT == *" Cassandra 3. "* ]]; then
            ALIVE = true
        else
            ALIVE =false
            fi
        else
            ALIVE =false
    fi
fi
```

**Comment by cscetbon** [comment]
The new proposal is to do the same as https://github.com/sky-uk/cassandra-operator/issues/30#issuecomment-480208420. Jolokia is running on each node and can be used to query the status. It's also really faster than using nodetool who spawn a new JVM each time it's called

**Comment by jsanda** [comment]
Very interesting. I think it is a neat approach.

**Comment by cscetbon** [comment]
Done and merged see #184
